internal/replacements/replace_failed_process_groups.go

```diff
@@ -92,7 +93,9 @@ func ReplaceFailedProcessGroups(log logr.Logger, cluster *fdbv1beta2.FoundationD
 		}
 
 		skipExclusion := false
-		if len(processGroupStatus.Addresses) == 0 {
+		// Only if localities are not used for exclusions we want to skip the exclusion. Skipping the exclusion could
+		// lead to a race condition, see: https://github.com/FoundationDB/fdb-kubernetes-operator/issues/1890
```

Comment:
If I understand correctly, the race condition can still occur here, it's just that it can be prevented when we are able to exclude by locality, right?  If true, I think it would be slightly clearer worded like this:
```suggestion
		// Only if localities are not used for exclusions do we want to skip the exclusion. Skipping the exclusion could
		// lead to a race condition, which can be prevented if we are able to exclude by locality.
		// see: https://github.com/FoundationDB/fdb-kubernetes-operator/issues/1890
```